src/Data/Generic/Rep/Enum.purs

```diff
@@ -0,0 +1,145 @@
+module Data.Generic.Rep.Enum
+  ( genericSucc
+  , class GenericSucc
+  , succ'
+
+  , genericPred
+  , class GenericPred
+  , pred'
+
+  , genericFromEnum
+  , class GenericFromEnum
+  , fromEnum'
+
+  , genericToEnum
+  , class GenericToEnum
+  , toEnum'
+
+  , genericCardinality
+  , class GenericCardinality
+  , cardinality'
+  ) where
+
+import Data.Generic.Rep
+import Data.Maybe (Maybe(..))
+import Data.Enum (Cardinality(Cardinality))
+import Data.Newtype (unwrap)
+import Prelude ((<$>), (+), (-))
+
+class GenericSucc a where
+  succ' :: a -> Maybe a
+
+instance genericSuccSumConstructor ::
```

Comment:
The cases here are a bit complicated. And why no instance for a single data constructor, or `NoConstructors`?

Is it not possible to just write instances for `Sum`, `Constructor` and `NoConstructors`?